stubs/tensorflow/tensorflow/keras/layers/experimental/preprocessing.pyi

```diff
@@ -0,0 +1,6 @@
+import abc
+
+from tensorflow._aliases import TensorLike
+from tensorflow.keras.layers import Layer
+
+class PreprocessingLayer(Layer[TensorLike, TensorLike], metaclass=abc.ABCMeta): ...
```

Comment:
Hmm, while Layer[TensorLike, TensorLike] is true I suspect it'll be a bit noisy and overloads for calls would be helpful here. I'd expect that a lot/most code that uses any of IndexLookup layers will be passing specific tensor type (like tf.Tensor) and returning TensorLike will make a bunch of false positives. Specifically something like,

```
    @overload
    def __call__(self, inputs: tf.Tensor) -> tf.Tensor: ...
    @overload
    def __call__(self, inputs: tf.SparseTensor) -> tf.SparseTensor: ...
    @overload
    def __call__(self, inputs: tf.RaggedTensor) -> tf.RaggedTensor: ...  # type: ignore
```

Other thing is major aspect of this class that differs it from normal layer is adapt/compile methods which I'd include [here](https://www.tensorflow.org/api_docs/python/tf/keras/layers/experimental/preprocessing/PreprocessingLayer#adapt)

Reply:
Done in 37c668ba1 and 4228893b9.

For the data arg, the doc says it must be a Tensor or a numpy array, but in the example they give it's a list, so I added `DataSequence` as an acceptable type.